src/rust/ensogl/src/display/shape/text/text_field.rs

```diff
@@ -126,6 +126,12 @@ shared! { TextField
     }
 
     impl {
+        /// Run `callback` with the underlying `&TextFieldContent`.
+        pub fn with_content<F:FnMut(&TextFieldContent)>(&self, callback:F) {
+            let mut callback = callback;
+            callback(&self.content);
+        }
+
```

Comment:
This function seems to be written to avoid restrictions of shared! macro... which restrictions were put there for reason. Now it is unsafe, because nobody is warned that callback cannot use any function of text_field, or we get double borrow panic.

I see you need it only for getting line height - please add function for getting height instead.

The same is for `with_mut_content`, however it has no simple replacement. It should at least has a comment which will warn about potential dangers.